src/main/java/com/cloudant/client/internal/views/PageMetadata.java

```diff
@@ -30,23 +43,79 @@
     /**
      * The parameters needed to request the page described by this metadata
      */
+    @Expose
+    @SerializedName("v")
     ViewQueryParameters<K, V> pageRequestParameters;
 
     /**
      * The page number of the page to retrieve
      */
+    @Expose
+    @SerializedName("n")
     long pageNumber;
     /**
      * Indicate the paging direction for this metadata
      */
+    @Expose
+    @SerializedName("d")
     PagingDirection direction;
 
     enum PagingDirection {
+        @SerializedName("F")
         FORWARD,
+        @SerializedName("B")
         BACKWARD
     }
 
     PageMetadata(PagingDirection direction) {
         this.direction = direction;
     }
+
+    static <K, V> PageMetadata<K, V> fromString(final ViewQueryParameters<K, V> initialParameters,
```

Comment:
Following this comment and our offline discussion about it I have isolated the paging information from the query parameters and created a `PaginationToken` class for serializing the necessary information for the opaque string pagination token. All in 63c295b, other than a quick method rename in 425bf37.